util-stats/src/main/scala/com/twitter/finagle/stats/StatsReceiver.scala

```diff
@@ -253,4 +256,19 @@ trait StatsReceiver {
   }
 }
 
-abstract class AbstractStatsReceiver extends StatsReceiver
+abstract class AbstractStatsReceiver extends StatsReceiver {
+  @varargs
+  final override def counter(name: String*): Counter = counter(Verbosity.Default, name: _*)
+  @varargs
+  final override def counter(verbosity: Verbosity, name: String*): Counter = counterImpl(verbosity, name)
+  protected def counterImpl(verbosity: Verbosity, name: scala.collection.Seq[String]): Counter
+  @varargs
+  final override def stat(name: String*): Stat = stat(Verbosity.Default, name: _*)
+  @varargs
+  final override def stat(verbosity: Verbosity, name: String*): Stat = statImpl(verbosity, name)
+  protected def statImpl(verbosity: Verbosity, name: scala.collection.Seq[String]): Stat
+  
+  final override def addGauge(name: String*)(f: => Float): Gauge = addGauge(Verbosity.Default, name: _*)(f)
```

Comment:
@varargs makes little sense if the last argument list isn't the varargs list. Shall I check if they actually generate any different bytecode?

Reply:
It does produce different bytecode.

The code

```
object Test {
  @varargs def withAnnotation(args: Int*)(extra: String) = args.length
}

```
produces extra an extra forwarder method:

```
  public int withAnnotation(int..., java.lang.String);
    descriptor: ([ILjava/lang/String;)I
    flags: ACC_PUBLIC, ACC_VARARGS
    Code:
      stack=3, locals=3, args_size=3
         0: aload_0
         1: getstatic     #31                 // Field scala/runtime/ScalaRunTime$.MODULE$:Lscala/runtime/ScalaRunTime$;
         4: aload_1
         5: invokevirtual #35                 // Method scala/runtime/ScalaRunTime$.wrapIntArray:([I)Lscala/collection/immutable/ArraySeq;
         8: aload_2
         9: invokevirtual #38                 // Method withAnnotation:(Lscala/collection/immutable/Seq;Ljava/lang/String;)I
        12: ireturn
```

But this can't ever be called with varargs from Java, since while this is valid JVM bytecode, there is no equivalent Java code, where the varargs must always be last.

In fact, I'd argue that the annotation here can't be useful, and should be deprecated in scala